topdown/cache/cache.go

```diff
@@ -110,6 +111,15 @@ func (c *cache) Delete(k ast.Value) {
 	c.unsafeDelete(k)
 }
 
+func (c *cache) UpdateConfig(config *Config) {
+	if config == nil {
+		return
+	}
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	c.limit = *config.InterQueryBuiltinCache.MaxSizeBytes
```

Comment:
Nit: Currently our cache config only contains one configurable parameter ie `size`. What if we maintain the config on the `cache` implementation and add a helper maybe to fetch the limit ? This would also make the function name more apt imo.

Reply:
I'm good with that